Refactor GroqModel to Data Class for Custom IDs (Syntax-Level Compatibility)

adamglin0 opened the pull request:
## Why

Enable custom GroqModel IDs. Backward compatibility is primarily syntax-level.

## What

Changed GroqModel from enum to data class. Retained existing constants (e.g., GroqModel.LLAMA_3_8B_8192) as pre-defined instances.

## How

Existing code using constants compiles. Important:  Address potential compatibility issues in serialization, configuration, API interactions, and type safety. Requires thorough testing and potential migration efforts.

adamglin0 committed:
Make GroqModel be data class

vyfor commented on src/commonMain/kotlin/io/github/vyfor/groqkt/GroqModel.kt:
```diff
@@ -2,55 +2,37 @@
 
-@Serializable(ModelSerializer::class)
-enum class GroqModel(
+@Serializable
```
```suggestion
@Serializable(ModelSerializer::class)
```

vyfor commented on src/commonMain/kotlin/io/github/vyfor/groqkt/GroqModel.kt:
```diff
@@ -2,55 +2,37 @@
+        )
+        val LLAVA_V1_5_7B_4096_PREVIEW = GroqModel("llava-v1.5-7b-4096-preview")
+    }
 }
```
```suggestion
}

private object ModelSerializer : KSerializer<GroqModel> {
    override val descriptor = PrimitiveSerialDescriptor("GroqModel", PrimitiveKind.STRING)

    override fun deserialize(decoder: Decoder): GroqModel {
        val id = decoder.decodeString()
        return GroqModel(id)
    }

    override fun serialize(encoder: Encoder, value: GroqModel) {
        encoder.encodeString(value.id)
    }
}
```

We probably still need this.

cn-tre-lfp replied:
OK, I think value classes can meet the requirements.

adamglin0 replied:
OK, I think value classes can meet the requirements.

vyfor reviewed (changes requested):
Looks good! I've eyeballed the necessary changes for now, will test everything later and update you.

adamglin0 committed:
Add tests for GroqModel serialization/deserialization

Introduce unit tests to ensure correct serialization and deserialization of the `GroqModel` class. Updated `GroqModel` to use a Kotlin `value class` for efficiency and modified Gradle configuration to centralize test dependencies in `commonTest`.

vyfor reviewed (approved):
LGTM! Thanks for adding the tests.
